### PR TITLE
Address yara-x compile findings

### PR DESCRIPTION
**Pull request by egibs**
Addresses: #638

Running `yr compile rules/` returned ~73 errors which were mostly duplicate rule names. Some of these rules were direct duplicates of each other; others were just needed a more appropriate (if longer) name.

In cases where `(elf or macho)` was used, the private rules needed to be renamed because the compiler thought we were referencing the import name (with a `struct` type).

Otherwise, I fixed a few strings and regenerated all of the test data to account for the updated rules.

To catch future rule warnings or errors, I added another CI check that will run `yr compile` on the entire rules directory which barely takes a second. If any findings exist, the Job will fail.

### CODE REVIEW

File: rules/impact/rootkit/rootkit.yara [hunk 1]
```diff
@@ -44,34 +44,3 @@ rule linux_kernel_module_hide_self: critical linux {
   condition:
     filesize < 1MB and $register_kprobe and any of ($hide*)
 }
-
-rule funky_high_signal_killer: high {
```
Comment (egibs) [inline rules/impact/rootkit/rootkit.yara:48]: This was a line-for-line duplicate.

File: rules/impact/rootkit/rootkit.yara [hunk 2]
```diff
@@ -44,34 +44,3 @@ rule linux_kernel_module_hide_self: critical linux {
...
-
-  strings:
-    $odd_teen_sig = /kill -1[012346789]/ fullword
-    $high_sig     = /kill -[23456]\d/ fullword
-
-  condition:
-    filesize < 10MB and any of them
-}
-
-rule linux_rootkit_terms: critical linux {
```
Comment (egibs) [inline rules/impact/rootkit/rootkit.yara:60]: This was a line-for-line duplicate.

File: rules/evasion/self_deletion/run_sleep_delete.yara [hunk 3]
```diff
@@ -29,15 +29,3 @@ rule fetch_run_sleep_delete: critical {
   condition:
     filesize < 1KB and $url and $sleep and $rm and any of ($path*) and any of ($run*)
 }
-
-rule self_delete: high {
```
Comment (egibs) [inline rules/evasion/self_deletion/run_sleep_delete.yara:33]: This was a line-for-line duplicate.

File: .github/workflows/style.yaml [hunk 4]
```diff
@@ -117,22 +141,3 @@ jobs:
           version: v1.62.0
           args: --timeout=5m
 
-  lint:
```
Comment (egibs) [inline .github/workflows/style.yaml:120]: This was doing nothing and we're already covering what it would handle via the other Jobs.

**Review by tstromberg — APPROVED** [review]

**Pull request merged**